### DISCUSSION TITLE
how to get TIMEOUT state from waitset.timedWait?

### DISCUSSION THREAD

**Discussion by leiBravo** [eclipse-iceoryx/iceoryx#2312] [Q&A] [answered]
hi, i am new to C++17. I am studying the request response example client_cxx_waitset.cpp. 
```
auto notificationVector = waitset.timedWait(iox::units::Duration::fromSeconds(5));
```
I want to know how to get the TIMEOUT state from waitset.timedWait?

**Comment by elBoberido** [comment, marked as answer]
Hi @leiBravo 

The timeout state is an empty `notificationVector` :)

```cpp
if(notificationVector.empty()) {
    std::cout << "The WaitSet timed out before notifications arrived" << std::endl;
}
```

**Comment by leiBravo** [comment]
Thank you for your answer, but I have already try this way, it’s doesn’t work, the size of vector can not indicate TIMEOUT

---- Replied Message ----
| From | Mathias ***@***.***> |
| Date | 07/05/2024 18:42 |
| To | eclipse-iceoryx/iceoryx ***@***.***> |
| Cc | leiBravo ***@***.***>,
Mention ***@***.***> |
| Subject | Re: [eclipse-iceoryx/iceoryx] how to get TIMEOUT state from waitset.timedWait? (Discussion #2312) |

Hi @leiBravo

The timeout state is an empty notificationVector :)

if(notificationVector.empty()) {
    std::cout << "The WaitSet timed out before notifications arrived" << std::endl;
}

**Reply by elBoberido** [reply]
Well, by definition the notification vector is empty on a timeout.

What did you try and did not work?

**Comment by leiBravo** [comment]
@elBoberido I have re-modified the test code, notificationVector.empty() is work. thanks
but I have another question. The Client send request to Server, The Server delay reply to client, then Client send the same request again. The Server reply a completely new response(new allocate memory address) ro just modify the first response ?

**Reply by elBoberido** [reply]
Well, that's up to how you implement the server. If you send responses to both requests, then both will arrive at the client. You need to adjust the sequence number to account for that. iceoryx does not handle the requests itself, it is only responsible for the transport and the user needs to decide how to handle these situations.

> I have re-modified the test code, notificationVector.empty() is work. thanks
I guess this discussion can be closed?
